src/util/memory.rs

```diff
@@ -144,24 +233,58 @@ pub fn dzmmap_noreplace(start: Address, size: usize, strategy: MmapStrategy) ->
 /// This function does not reserve swap space for this mapping, which means there is no guarantee that writes to the
 /// mapping can always be successful. In case of out of physical memory, one may get a segfault for writing to the mapping.
 /// We can use this to reserve the address range, and then later overwrites the mapping with dzmmap().
-pub fn mmap_noreserve(start: Address, size: usize, mut strategy: MmapStrategy) -> Result<()> {
+pub fn mmap_noreserve(
+    start: Address,
+    size: usize,
+    mut strategy: MmapStrategy,
+    anno: &MmapAnnotation,
+) -> Result<()> {
     strategy.prot = MmapProtection::NoAccess;
     let flags = MMAP_FLAGS | libc::MAP_NORESERVE;
-    mmap_fixed(start, size, flags, strategy)
+    mmap_fixed(start, size, flags, strategy, anno)
 }
 
 fn mmap_fixed(
     start: Address,
     size: usize,
     flags: libc::c_int,
     strategy: MmapStrategy,
+    _anno: &MmapAnnotation,
 ) -> Result<()> {
     let ptr = start.to_mut_ptr();
     let prot = strategy.prot.into_native_flags();
     wrap_libc_call(
         &|| unsafe { libc::mmap(start.to_mut_ptr(), size, prot, flags, -1, 0) },
         ptr,
     )?;
+
+    #[cfg(target_os = "linux")]
```

Comment:
```suggestion
    #[cfg(any(target_os = "linux", target_os = "android"))]
```

Reply:
Tell me about it. My fork of MMTk core is riddled with this stuff so that it works with Android as well. I honestly don't get it either.